source/windows/cross_process_lock.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <windows.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/cross_process_lock.h>
+#include <aws/common/error.h>
+#include <aws/common/logging.h>
+#include <inttypes.h>
+
+struct aws_cross_process_lock {
+    struct aws_allocator *allocator;
+    HANDLE mutex;
+};
+
+struct aws_cross_process_lock *aws_cross_process_lock_try_acquire(
+    struct aws_allocator *allocator,
+    struct aws_byte_cursor instance_nonce) {
+
+    /* validate we don't have a directory slash. */
+    struct aws_byte_cursor to_find = aws_byte_cursor_from_c_str("\\");
+    struct aws_byte_cursor found;
+    AWS_ZERO_STRUCT(found);
+    if (aws_byte_cursor_find_exact(&instance_nonce, &to_find, &found) != AWS_OP_ERR &&
+        aws_last_error() != AWS_ERROR_STRING_MATCH_NOT_FOUND) {
+        AWS_LOGF_ERROR(
+            AWS_LS_COMMON_GENERAL,
+            "static: Lock " PRInSTR " creation has illegal character \\",
+            AWS_BYTE_CURSOR_PRI(instance_nonce));
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
+    }
+
+    struct aws_cross_process_lock *instance_lock = NULL;
+
+    /* Local prefix, per the docs, specifies session scope rather than global to the user or system. */
+    struct aws_byte_cursor path_prefix = aws_byte_cursor_from_c_str("Local/aws_crt_cross_process_lock/");
```

Comment:
"Local\\" with backslash. It's not clear in the CreateMutexA() docs, but they link [other docs](https://learn.microsoft.com/en-us/windows/win32/termserv/kernel-object-namespaces) that show it
```suggestion
    /* "Local\" prefix, per the docs, specifies user session scope (rather than "Global\" to the system). */
    struct aws_byte_cursor path_prefix = aws_byte_cursor_from_c_str("Local\\aws_crt_cross_process_lock/");
```

Reply:
thank you!